crates/flowctl/src/local_specs.rs

```diff
@@ -126,6 +125,35 @@ pub(crate) async fn inline_and_validate(
         })
         .collect::<tables::Errors>();
 
+    // Gather all files generated by validations.
+    let mut generated_files = BTreeMap::new();
+    for row in validations.built_collections.iter() {
+        let Some(validated) = &row.validated else { continue };
+        for (url, content) in &validated.generated_files {
+            match url::Url::parse(&url) {
+                Ok(url) => {
+                    generated_files.insert(url, content.as_bytes());
+                }
+                Err(err) => errors.insert_row(
+                    &row.scope,
+                    anyhow::anyhow!(err)
+                        .context("derive connector returns invalid generated file URL"),
+                ),
+            }
+        }
+    }
+
+    // Write out all generated files.
+    if let Err(error) = write_files(
```

Comment:
It seems like the name and the body of this function have diverged significantly. Rename?